src/ASM/GlbL2projector.C

```diff
@@ -33,6 +34,102 @@ LinAlg::MatrixType GlbL2::MatrixType   = LinAlg::SPARSE;
 LinSolParams*      GlbL2::SolverParams = nullptr;
 
 
+static void expandTensorGrid2 (const RealArray* in, RealArray* out)
+{
+  out[0].resize(in[0].size()*in[1].size());
+  out[1].resize(in[0].size()*in[1].size());
+
+  size_t i, j, ip = 0;
+  for (j = 0; j < in[1].size(); j++)
+    for (i = 0; i < in[0].size(); i++, ip++) {
+      out[0][ip] = in[0][i];
+      out[1][ip] = in[1][j];
+    }
+}
+
+
+static void expandTensorGrid3 (const RealArray* in, RealArray* out)
+{
+  out[0].resize(in[0].size()*in[1].size()*in[2].size());
+  out[1].resize(in[0].size()*in[1].size()*in[2].size());
+  out[2].resize(in[0].size()*in[1].size()*in[2].size());
+
+  size_t i, j, k, ip = 0;
+  for (k = 0; k < in[2].size(); k++)
+    for (j = 0; j < in[1].size(); j++)
+      for (i = 0; i < in[0].size(); i++, ip++) {
+        out[0][ip] = in[0][i];
+        out[1][ip] = in[1][j];
+        out[2][ip] = in[2][k];
+      }
+}
+
+
+L2ProbIntegrand::L2ProbIntegrand (const ASMbase& ASM, const IntegrandBase& itg) :
```

Comment:
I prefer to use `patch` or `pch` as variable name her for the AMSbase object. ASM is already defined as a namespace elsewhere and could be potentially confusing in the doxy processing (or only in my head...)

Reply:
patch and m_patch.